src/main/java/com/joyent/manta/cosbench/MantaStorage.java

```diff
@@ -241,9 +265,102 @@ public void createObject(final String container,
+        }
+        logger.info("Object Size : " + this.objectSize);
+        logger.info("Split Size : " + splitSize);
+        int splits = objectSize / splitSize;
+        int remainder = objectSize % splitSize;
+        if (remainder > 0) {
+            splits++;
+        }
+        MantaMultipartUploadTuple[] parts = new MantaMultipartUploadTuple[splits];
+        // ServerSideMultipartUpload upload =
+        // multipartManager.initiateUpload(path);
+        ServerSideMultipartUpload upload = multipartManager.initiateUpload(path);
+
+        logger.debug("Multipart upload initiated");
+        byte[] content = IOUtils.toByteArray(data);
+        for (int i = 0; i < splits; i++) {
+            int startIndex = i * splitSize;
+            int endIndex;
+            // Parts start at 1.
+            int partNumber = i + 1;
+            if (((i + 1) * splitSize) < content.length) {
+                endIndex = partNumber * splitSize;
+            } else {
+                endIndex = content.length;
+            }
+            logger.info("Part Number {}", partNumber);
+            logger.info("Starting copy {} ending copy {}", startIndex, endIndex);
+            byte[] partData = Arrays.copyOfRange(content, startIndex, endIndex);
+            parts[i] = multipartManager.uploadPart(upload, partNumber, partData);
+            logger.debug("grabbed the next " + splitSize + " bytes");
+        }
+        Stream<MantaMultipartUploadTuple> partsStream = Arrays.stream(parts);
+        multipartManager.complete(upload, partsStream);
+        logger.debug(path + " is now assembled and sent as a multipart stream!");
+    }
+
+    /**
+     * This will take the stream parse it into sizes defined in the config.
+     * @param data - the data stream to be parsed.
+     * @param path - the path where this will be put.
+     * @throws IOException - this is a possibility.
+     */
+    private void multipartEncrypt(final InputStream data, final String path) throws IOException {
+        logger.info("Client is encrypted!");
+        EncryptedServerSideMultipartManager multipartManager = new EncryptedServerSideMultipartManager(client);
+        EncryptedMultipartUpload<ServerSideMultipartUpload> upload = multipartManager.initiateUpload(path);
+        logger.debug("Client is not encrypted!");
+        if (splitSize == null) {
+            splitSize = new Integer(DEFAULT_SPLIT);
+        }
+        logger.info("Object Size : " + this.objectSize);
+        logger.info("Split Size : " + splitSize);
+        int splits = objectSize / splitSize;
+        int remainder = objectSize % splitSize;
+        if (remainder > 0) {
+            splits++;
+        }
+        MantaMultipartUploadTuple[] parts = new MantaMultipartUploadTuple[splits];
+        // ServerSideMultipartUpload upload =
+        // multipartManager.initiateUpload(path);
+        logger.debug("Multipart upload initiated");
+        byte[] content = IOUtils.toByteArray(data);
```

Comment:
Whatever we land on for this logic, it should be broken into a separate method in order to avoid duplication.